components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt

```diff
@@ -76,36 +90,45 @@ class FxaPushSupportFeature(
  */
 internal class AccountObserver(
     private val context: Context,
-    private val accountManager: FxaAccountManager,
     private val push: AutoPushFeature,
+    private val pushScope: String,
     private val lifecycleOwner: LifecycleOwner,
     private val autoPause: Boolean
 ) : SyncAccountObserver {
 
-    private val logger = Logger("AccountObserver")
+    private val logger = Logger(AccountObserver::class.java.simpleName)
     private val constellationObserver = ConstellationObserver(context, push)
 
     override fun onAuthenticated(account: OAuthAccount, authType: AuthType) {
         // We need a new subscription only when we have a new account.
         // The subscription is removed when an account logs out.
         if (authType != AuthType.Existing && authType != AuthType.Recovered) {
-            logger.debug("Subscribing for ${PushType.Services} events.")
+            logger.debug("Subscribing for $pushScope events.")
 
-            push.subscribeForType(PushType.Services)
+            push.subscribe(pushScope) { subscription ->
+                account.deviceConstellation().setDevicePushSubscriptionAsync(
+                    DevicePushSubscription(
+                        endpoint = subscription.endpoint,
+                        publicKey = subscription.publicKey,
+                        authKey = subscription.authKey
+                    )
+                )
+            }
         }
 
-        accountManager.withConstellation { constellation ->
-            constellation.registerDeviceObserver(constellationObserver, lifecycleOwner, autoPause)
-        }
+        account.deviceConstellation().registerDeviceObserver(constellationObserver, lifecycleOwner, autoPause)
     }
 
     override fun onLoggedOut() {
-        logger.debug("Un-subscribing for ${PushType.Services} events.")
+        logger.debug("Un-subscribing for FxA scope $pushScope events.")
 
-        push.unsubscribeForType(PushType.Services)
+        push.unsubscribe(pushScope)
 
-        // Delete cached value of last verified timestamp when we log out.
-        preference(context).edit().remove(LAST_VERIFIED).apply()
+        // Delete cached value of last verified timestamp and scope when we log out.
+        preference(context).edit()
+            .remove(PREF_LAST_VERIFIED)
+            .remove(PREF_FXA_SCOPE)
```

Comment:
Same comment above, there's no strong reasoning to, but it seems odd if every FxA user had the exact same subscription ID for FxA.

> why is it important to erase the scope value?

If we stick with that reasoning, removal is so that a new user doesn't have the same subscription ID as the previous user.

Reply:
Why is that weird? It seems fine to me. As mentioned above, this also matches semantics of webpush. Each user has a unique UAID, and with that UAID you may associate "fxa channel id", "cnn.com channel id", etc. These will all end up as unique endpoints exposed by autopush.